code/__DEFINES/flags.dm

```diff
@@ -15,3 +15,11 @@ GLOBAL_LIST_INIT(bitflags, list(1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 204
 #define DF_USE_TAG		(1<<0)
 #define DF_VAR_EDITED	(1<<1)
 #define DF_ISPROCESSING (1<<2)
+
+//Movement Types
+#define GROUND			(1<<0)
+#define FLYING			(1<<1)
+#define VENTCRAWLING	(1<<2)
+#define FLOATING		(1<<3)
```

Comment:
How is floating different from flying?

Reply:
Floating is from lack of gravity. Flying is self-powered floating with more control, I guess.

Reply:
I guess flying is also higher? Like, you won't fall over a railing if you are floating in no gravity unless you purposely jump over, but you can fly over it.
The difference is not huge and clearly defined, granted, as z-levels and stuff are not yet very well consolidated. But this is how /tg/ does it currently.

Reply:
Im just looking to get them defined so we know how we want to apply them.,

For example i'd see floating as no higher than normal.
They would never slip or touch objects on the floor, but they wouldn't float above things like windows / tables etc

Reply:
Sounds fair to me.